src/equations/ideal_glm_mhd_multicomponent_1d.jl

```diff
@@ -0,0 +1,469 @@
+
+@doc raw"""
+    IdealGlmMhdEquations1D
+
+The ideal compressible multicomponent GLM-MHD equations in one space dimension.
+"""
+mutable struct IdealGlmMhdMulticomponentEquations1D{NVARS, NCOMP, RealT<:Real} <: AbstractIdealGlmMhdMulticomponentEquations{1, NVARS, NCOMP}
+  gammas            ::SVector{NCOMP, RealT}
+  gas_constants     ::SVector{NCOMP, RealT}
+  cv                ::SVector{NCOMP, RealT}
+  cp                ::SVector{NCOMP, RealT}
+
+  function IdealGlmMhdMulticomponentEquations1D{NVARS, NCOMP, RealT}(gammas       ::SVector{NCOMP, RealT},
+                                                                     gas_constants::SVector{NCOMP, RealT}) where {NVARS, NCOMP, RealT<:Real}
+
+    NCOMP >= 1 || throw(DimensionMismatch("`gammas` and `gas_constants` have to be filled with at least one value"))
+
+    cv = gas_constants ./ (gammas .- 1)
+    cp = gas_constants + gas_constants ./ (gammas .- 1)
+
+    new(gammas, gas_constants, cv, cp)
+  end
+end
+
+function IdealGlmMhdMulticomponentEquations1D(; gammas, gas_constants)
+
+  _gammas        = promote(gammas...)
+  _gas_constants = promote(gas_constants...)
+  RealT          = promote_type(eltype(_gammas), eltype(_gas_constants))
+
+  NVARS = length(_gammas) + 7
+  NCOMP = length(_gammas)
+
+  __gammas        = SVector(map(RealT, _gammas))
+  __gas_constants = SVector(map(RealT, _gas_constants))
```

Comment:
No need to change this here, but please be aware of the [naming conventations](https://trixi-framework.github.io/Trixi.jl/stable/conventions/#Variable-names) in that shadowing should be avoided by an underscore suffix, while an underscore prefix is used to indicate internal/fragile data or implementations.